controllers/disruption_controller.go

```diff
@@ -343,91 +531,128 @@ func (r *DisruptionReconciler) updateInjectionStatus(instance *chaosv1beta1.Disr
 
 // startInjection creates non-existing chaos pod for the given disruption
 func (r *DisruptionReconciler) startInjection(instance *chaosv1beta1.Disruption) error {
-	var err error
-
 	if len(instance.Status.Targets) > 0 {
 		r.log.Infow("starting targets injection", "targets", instance.Status.Targets)
 	}
 
+	// chaosPodsMap will be used to check if a target has an existing chaos pod already or not
+	chaosPodsMap := map[string]corev1.Pod{}
+
+	chaosPods, err := r.getChaosPods(instance, nil)
+	if err != nil {
+		return fmt.Errorf("error getting chaos pods: %w", err)
+	}
+
+	for _, chaosPod := range chaosPods {
+		if !utils.Contains(instance.Status.Targets, chaosPod.Labels[chaostypes.TargetLabel]) {
+			r.deleteChaosPod(instance, chaosPod)
+		} else {
+			chaosPodsMap[chaosPod.Labels[chaostypes.TargetLabel]+"-"+chaosPod.Labels[chaostypes.DisruptionKindLabel]] = chaosPod
```

Comment:
What do you think about changing `chaosPodsMap` to be `map[string]map[string]corev1.Pod{}` where the inner key is the kind, so that we aren't using composite keys like this?

Reply:
It also seems like we don't ever check the value of this map? We can just use `bool` in that case, for "it exists", if that's all we really care about, rather than store whole Pods (we're not even storing pointers)